server/migration.sql

```diff
@@ -25,3 +26,7 @@ CREATE TABLE chat_history(
     ai_message TEXT NOT NULL,
     created_at TIMESTAMP NOT NULL);
 
+CREATE TABLE users(
+    id SERIAL NOT NULL PRIMARY KEY,
```

Comment:
You can remove NOT NULLfrom any id columns in this file. PRIMARY KEY by default should take care of NOT NULL.